analyzer/consensus/consensus.go

```diff
@@ -801,103 +889,139 @@ func (m *Main) queueVotes(batch *storage.QueryBatch, data *storage.GovernanceDat
 	case e.Governance != nil:
-		backend = analyzer.BackendGovernance
-		switch b := event.Governance; {
-		case b.ProposalSubmitted != nil:
-			ty = analyzer.EventGovernanceProposalSubmitted
-			body, err = json.Marshal(event.Governance.ProposalSubmitted)
-			return
-		case b.ProposalExecuted != nil:
-			ty = analyzer.EventGovernanceProposalExecuted
-			body, err = json.Marshal(event.Governance.ProposalExecuted)
-			return
-		case b.ProposalFinalized != nil:
-			ty = analyzer.EventGovernanceProposalExecuted
-			body, err = json.Marshal(event.Governance.ProposalFinalized)
-			return
-		case b.Vote != nil:
-			ty = analyzer.EventGovernanceVote
-			body, err = json.Marshal(event.Governance.Vote)
-			return
-		}
+		return extractGovernanceEvent(event.Governance)
+	}
+
+	return analyzer.EventUnknown, []byte{}, errors.New("unknown event type")
+}
+
+func extractGovernanceEvent(event *governance.Event) (analyzer.Event, []byte, error) {
+	var ty analyzer.Event
+	var body []byte
+	var err error
+	switch event := event; {
+	case event.ProposalSubmitted != nil:
+		ty = analyzer.EventGovernanceProposalSubmitted
+		body, err = json.Marshal(event.ProposalSubmitted)
+	case event.ProposalExecuted != nil:
+		ty = analyzer.EventGovernanceProposalExecuted
+		body, err = json.Marshal(event.ProposalExecuted)
+	case event.ProposalFinalized != nil:
+		ty = analyzer.EventGovernanceProposalExecuted
+		body, err = json.Marshal(event.ProposalFinalized)
+	case event.Vote != nil:
+		ty = analyzer.EventGovernanceVote
+		body, err = json.Marshal(event.Vote)
+	}
+	return ty, body, err
+}
+
+func extractRootHashEvent(event *roothash.Event) (analyzer.Event, []byte, error) {
+	var ty analyzer.Event
+	var body []byte
+	var err error
+	switch event := event; {
+	case event.ExecutorCommitted != nil:
+		ty = analyzer.EventRoothashExecutorCommitted
+		body, err = json.Marshal(event.ExecutorCommitted)
+	case event.ExecutionDiscrepancyDetected != nil:
+		ty = analyzer.EventRoothashExecutionDiscrepancyDetected
+		body, err = json.Marshal(event.ExecutionDiscrepancyDetected)
+	case event.Finalized != nil:
+		ty = analyzer.EventRoothashFinalized
+		body, err = json.Marshal(event.Finalized)
 	}
+	return ty, body, err
+}
 
-	return analyzer.BackendUnknown, analyzer.EventUnknown, []byte{}, errors.New("unknown event type")
+func extractRegistryEvent(event *registry.Event) (analyzer.Event, []byte, error) {
+	var ty analyzer.Event
+	var body []byte
+	var err error
+	switch event := event; {
+	case event.RuntimeEvent != nil:
+		ty = analyzer.EventRegistryRuntime
+		body, err = json.Marshal(event.RuntimeEvent)
+	case event.EntityEvent != nil:
+		ty = analyzer.EventRegistryEntity
+		body, err = json.Marshal(event.EntityEvent)
+	case event.NodeEvent != nil:
+		ty = analyzer.EventRegistryNode
+		body, err = json.Marshal(event.NodeEvent)
+	case event.NodeUnfrozenEvent != nil:
+		ty = analyzer.EventRegistryNodeUnfrozen
+		body, err = json.Marshal(event.NodeUnfrozenEvent)
+	}
```

Comment:
Let's be defensive here and in all the other `extract*Event` functions:

```suggestion
  default:
    return fmt.Errorf("unsupported registry event type: %#v", event)
	}
```